contracts/src/StakeTable.sol

```diff
@@ -52,493 +102,352 @@ contract StakeTable is AbstractStakeTable, Ownable, InitializedAt {
         }
 
-        if (node.account != msg.sender) {
-            revert Unauthenticated();
+        uint256 balance = delegations[validator][msg.sender];
+        if (balance < amount) {
+            revert InsufficientBalance(balance);
         }
 
-        // Cannot request to exit if an exit request is already in progress.
-        if (node.exitEpoch != 0) {
-            revert ExitRequestInProgress();
+        delegations[validator][msg.sender] -= amount;
+        undelegations[validator][msg.sender] =
+            Undelegation({ amount: amount, unlocksAt: block.timestamp + exitEscrowPeriod });
+
+        emit Undelegated(msg.sender, validator, amount);
+    }
+
+    /// @notice Withdraw previously delegated funds after an undelegation.
+    /// @param validator The validator to withdraw from
+    function claimWithdrawal(address validator) external virtual {
+        // If entries are missing at any of the levels of the mapping this will return zero
+        uint256 amount = undelegations[validator][msg.sender].amount;
+        if (amount == 0) {
+            revert NothingToWithdraw();
         }
 
-        // Cannot exit before becoming an active participant. Activation happens one epoch after the
-        // node's registration epoch, due to the consensus-imposed activation waiting period.
-        if (currentEpoch() < node.registerEpoch + 1) {
-            revert PrematureExit();
+        if (block.timestamp < undelegations[validator][msg.sender].unlocksAt) {
+            revert PrematureWithdrawal();
         }
 
-        // Prepare the node to exit.
-        pushToExitQueue();
-        nodes[msg.sender].exitEpoch = exitEpoch;
+        // Mark funds as spent
+        delete undelegations[validator][msg.sender];
 
-        emit Exit(msg.sender, exitEpoch);
-    }
+        SafeTransferLib.safeTransfer(token, msg.sender, amount);
 
-    /// @notice Withdraw from the staking pool. Transfers occur! Only successfully exited keys can
-    /// withdraw past their `exitEpoch`. Validators have to first call requestExit to be assigned an
-    /// exit epoch
-    /// @return The total amount withdrawn, equal to `Node.balance` associated with `blsVK`
-    /// TODO: add epoch logic so that we can ensure the node has first requested to exit and waiting
-    /// for the exit escrow period to be over
-    function withdrawFunds() external virtual override returns (uint256) {
-        Node memory node = nodes[msg.sender];
-
-        if (node.account == address(0)) {
-            revert NodeNotRegistered();
-        }
+        emit Withdrawal(msg.sender, amount);
+    }
 
-        if (node.account != msg.sender) {
-            revert Unauthenticated();
+    /// @notice Withdraw previously delegated funds after a validator has exited
+    /// @param validator The validator to withdraw from
+    function claimValidatorExit(address validator) external virtual {
+        uint256 unlocksAt = validatorExits[msg.sender];
```

Comment:
`validatorExits` is only updated when the `deregisterValidator()` function is called so does that mean this function is only to be called by validators? if so, it's not immediately obvious with the naming

Reply:
this is indeed wrong, fixed in https://github.com/EspressoSystems/espresso-network/pull/2662/commits/fc3b31c9ff8c5d6616d816f6b6fd6176ba076331